src/iterator.rs

```diff
@@ -303,9 +554,16 @@ impl Signals {
     ///
     /// Note that the blocking is done in this method, not in the iterator.
     pub fn wait(&self) -> Pending {
-        self.flush(true);
-
-        Pending(self.waker.pending.iter().enumerate())
+        match self.0.wait(&mut Self::has_signals) {
+            Ok(Some(pending)) => pending,
+            Ok(None) => unreachable!(
+                "Because of the blocking has_signals method the next_signal \
+                method never returns Poll::Pending but blocks until a signal arrived"
+            ),
+            // Users can't manipulate the internal file descriptors and the way we use them
+            // shouldn't produce any errors. So it is OK to panic.
+            Err(error) => panic!("Unexpected error: {}", error),
```

Comment:
Hmm, are you sure nothing weird can happen, like out-of-memory error? I'm not sure it should be propagated, because the OS is probably very ill at that point and what should users do about the error…

Reply:
At least with respect to the [list of error kinds](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variants) we shouldn't get any errors. Most of the connection related errors can't happen because at this point we already have a pair of streams and will never close them. BrokenPipe shouldn't happen because the reading end of the UDS pipe won't be closed. The WouldBlock can't happen because we leave the streams always in blocking mode. An Interrupted error is handled in the has_signal method.

I'm not sure if I understand your second sentence right. Are you for or against panicing?

Reply:
Well, reading the man pages of low-level `recv`, it notes „Additional errors may be generated and returned from the underlying protocol modules“. In that sense, it seems the set of errors that can be returned is open to whatever errors. They would likely fall into the `ErrorKind::Other`. So in theory the `recv` could generate something like `ENOMEM` or `EIO`.

However, that worry is probably just me worrying too much and I would not expect them to happen. Therefore panicking is probably the right solution ‒ as these errors likely won't happen and if they do happen, it's because the OS is in some kind of already unusable state.